File: dataModule/src/main/java/com/mobnetic/coinguardian/model/CurrencyPairInfo.java
package com.mobnetic.coinguardian.model;


public class CurrencyPairInfo implements Comparable<CurrencyPairInfo>{
	
	protected final String currencyBase;
	protected final String currencyCounter;
	protected final String currencyPairId;
	
	public CurrencyPairInfo(String currencyBase, String currencyCounter, String currencyPairId) {
		this.currencyBase = currencyBase;
		this.currencyCounter = currencyCounter;
		this.currencyPairId = currencyPairId;
	}
	
	public String getCurrencyBase() {
		return currencyBase;
	}
	
	public String getCurrencyCounter() {
		return currencyCounter;
	}
	
	public String getCurrencyPairId() {
		return currencyPairId;
	}
	
	@Override
	public int compareTo(CurrencyPairInfo another) throws NullPointerException {
		if(currencyBase==null || another.currencyBase==null || currencyCounter==null || another.currencyCounter==null)
			throw new NullPointerException();
		
		int compBase = currencyBase.compareToIgnoreCase(another.currencyBase);
		return compBase!=0 ? compBase : currencyCounter.compareToIgnoreCase(another.currencyCounter);
	}
}

File: dataModule/src/main/java/com/aneonex/bitcoinchecker/datamodule/model/CurrencyPairInfo.kt
package com.aneonex.bitcoinchecker.datamodule.model

open class CurrencyPairInfo(
    val currencyBase: String,
    val currencyCounter: String,
    val currencyPairId: String?,
    val contractType: FuturesContractType = FuturesContractType.NONE
) : Comparable<CurrencyPairInfo> {

    @Suppress("unused") // Used by Gson
    private constructor() : this("", "", null)

    @Throws(NullPointerException::class)
    override fun compareTo(other: CurrencyPairInfo): Int {
        var compBase = currencyBase.compareTo(other.currencyBase, ignoreCase = true)
        if (compBase != 0) return compBase

        compBase = currencyCounter.compareTo(
            other.currencyCounter,
            ignoreCase = true
        )
        if (compBase != 0) return compBase

        return contractType.compareTo(other.contractType)
    }

    override fun toString(): String {
        fun tryGetContactName(): String {
            val resultName = FuturesContractType.getShortName(contractType)
            return if(resultName == null) "" else ":$resultName"
        }

        return currencyPairId ?:
        "$currencyBase:$currencyCounter" + tryGetContactName()
    }
}